plugins/pushbullet.js

```diff
@@ -1,88 +1,186 @@
+  }
+}
+
+Pushbullet.prototype.logRoundtripPart = function(trade) {
+  // this is not part of a valid roundtrip
+  if(this.trades === 1 && trade.action === 'sell') {
+    var body = 'Previous trade wass not part of a round-trip. Round-trip tracking will start on next trade.';
+    this.notify(this.title, body);
+    return;
+  }
+
+  if(trade.action === 'buy') {
+    this.roundTrip.entry = {
+      date: trade.date,
+      price: trade.price,
+      total: (trade.portfolio.asset * trade.price) + trade.portfolio.currency
+    }
+  } else if(trade.action === 'sell') {
+    this.roundTrip.exit = {
+      date: trade.date,
+      price: trade.price,
+      total: trade.portfolio.currency + (trade.portfolio.asset * trade.price)
+    }
+    this.handleRoundtrip();
+  }
+}
+  
+Pushbullet.prototype.handleRoundtrip = function() {
+  var roundtrip = {
+    entryAt: this.roundTrip.entry.date,
+    entryPrice: this.roundTrip.entry.price,
+    entryBalance: this.roundTrip.entry.total,
 
-Pushbullet.prototype.checkResults = function(err) {
-    if(err)
-        log.warn('error sending email', err);
-    else
-        log.info('Send advice via email.');
+    exitAt: this.roundTrip.exit.date,
+    exitPrice: this.roundTrip.exit.price,
+    exitBalance: this.roundTrip.exit.total,
+
+    duration: (moment(this.roundTrip.exit.date).diff(this.roundTrip.entry.date, 'seconds'))
+  }
+  
+  roundtrip.pnl = roundtrip.exitBalance - roundtrip.entryBalance;
+  roundtrip.profit = (100 * roundtrip.exitBalance / roundtrip.entryBalance) - 100;
+
+  var startBalance = this.start.currency + (this.startPrice * this.start.asset);
+  var balance = this.current.currency + (this.endPrice * this.current.asset);
+  var pTotal = (balance * 100 / startBalance - 100);
+  var market = (this.endPrice * 100 / this.startPrice - 100);
+
+  var subject = this.title + ' Round Trip Complete!';
+  var body = '\nEntry Balance: ' + roundtrip.entryBalance.toFixed(this.digitsCurrency) + this.currency;
+  body += '\nExit Balance: ' + roundtrip.exitBalance.toFixed(this.digitsCurrency) + this.currency;
+  body += '\nProfit: ' + roundtrip.profit.toFixed(2) + '% (' + roundtrip.pnl.toFixed(this.digitsCurrency) + this.currency + ')' ;
+  body += '\nExposed Time: ' + secToTimeString(roundtrip.duration);
+  body += '\nProfit Since Start: ' + pTotal.toFixed(2) + '%';
+  body += '\nMarket Performance: ' + market.toFixed(2) + '%';
+
+  this.notify(subject, body);
+}
+  
+Pushbullet.prototype.notify = function(subject, content, done) {
+  var pusher = new pushbullet(config.pushbullet.key);
+  pusher.note(config.pushbullet.email, subject, content, function(error, response) {
+    if(error || !response) {
+        log.error('Pushbullet ERROR:', error)
+    } else if(response && response.active){
+        log.info('Pushbullet Message Sent')
+    }
+  });
 };
 
-module.exports = Pushbullet;
\ No newline at end of file
+module.exports = Pushbullet;
+
+function secToTimeString(duration) {
```

Comment:
you can also use moment for this: http://momentjs.com/docs/#/durations/

Reply:
Yes, although I am unable to understand if the duration or duration.humanize() method can split it into multiple units of time like X hours Y minutes Z seconds, or if it can only work against 1 unit of measurement so I made a little helper function.